### PR TITLE
refactor: rename "noProxy" to "onlyNonProxyOwner"

### LINKED ISSUE #156: Refactor "noProxy(owner)" to "onlyNonProxyOwner(user)"
This is @IaroslavMazur's recommendation:

> `noProxy` would be better named to `onlyNonOwner` or `onlyNonProxyOwner`, while its argument would better be named just `user`; if the passed-in argument is known to be an owner when the modifier is called, there is no reason to call the modifier, at all.

I like `onlyNonProxyOwner`, and I also agree that `user` would be clearer.

### PR DESCRIPTION
**Pull request by PaulRBerg**
Closes #156.

### CODE REVIEW

File: src/interfaces/IPRBProxyRegistry.sol [hunk 1]
```diff
@@ -31,6 +25,12 @@ interface IPRBProxyRegistry {
     /// @notice Thrown when trying to install a plugin that doesn't implement any method.
     error PRBProxyRegistry_PluginWithZeroMethods(IPRBProxyPlugin plugin);
 
+    /// @notice Thrown when a function requires the user to have a proxy.
+    error PRBProxyRegistry_UserDoesNotHaveProxy(address caller);
+
+    /// @notice Thrown when a function requires the user to not have a proxy.
+    error PRBProxyRegistry_UserHasProxy(address owner, IPRBProxy proxy);
```
Comment (IaroslavMazur) [inline src/interfaces/IPRBProxyRegistry.sol:32]:
```suggestion
    error PRBProxyRegistry_UserHasProxy(address user, IPRBProxy proxy);
```

File: src/interfaces/IPRBProxyRegistry.sol [hunk 2]
```diff
@@ -31,6 +25,12 @@ interface IPRBProxyRegistry {
     /// @notice Thrown when trying to install a plugin that doesn't implement any method.
     error PRBProxyRegistry_PluginWithZeroMethods(IPRBProxyPlugin plugin);
 
+    /// @notice Thrown when a function requires the user to have a proxy.
+    error PRBProxyRegistry_UserDoesNotHaveProxy(address caller);
```
Comment (IaroslavMazur) [inline src/interfaces/IPRBProxyRegistry.sol:29]:
```suggestion
    error PRBProxyRegistry_UserDoesNotHaveProxy(address user);
```

**Review by IaroslavMazur — APPROVED** [review]

**Comment by IaroslavMazur** [comment]
@PaulRBerg, aside from the 2 minor suggestions, I see no other issues with this PR.

**Comment by PaulRBerg** [comment]
Thanks, @IaroslavMazur - I have made a few changes here and there in my latest commit. Can you have another final look, please?

**Comment by IaroslavMazur** [comment]
> Thanks, @IaroslavMazur - I have made a few changes here and there in my latest commit. Can you have another final look, please?

Went through the 4 new commits. Everything seems alright. 👍

**Pull request merged**